src/Resolvers/Microsoft.DotNet.MSBuildSdkResolver/MSBuildSdkResolver.cs

```diff
@@ -190,6 +191,22 @@ private sealed class CachedState
                         minimumVSDefinedSDKVersion);
                 }
 
+                string dotnetExe = Path.Combine(dotnetRoot, Constants.DotNet + Constants.ExeSuffix);
+                if (File.Exists(dotnetExe))
+                {
+                    propertiesToAdd ??= new Dictionary<string, string?>();
+                    propertiesToAdd.Add("DOTNET_HOST_PATH", dotnetExe);
```

Comment:
I feel medium-strongly that `DOTNET_HOST_PATH` shouldn't be the property name, since it's against convention. Maybe
```suggestion
                    propertiesToAdd.Add("SdkResolverDotnetHostPath", dotnetExe);
```
to match the `global.json` reports?

Reply:
One of the reasons we talked about using the DOTNET_HOST_PATH property was to take advantage of all of the existing logic that already relied on DOTNET_HOST_PATH if it was set - if we change the name we'd either need

* an early sdk-level mapping in the props from SdkResolverDotnetHostPath => DOTNET_HOST_PATH, or
* each of the current consumers of DOTNET_HOST_PATH to do this same mapping

Reply:
One thing to mention - in "dotnet build" case `DOTNET_HOST_PATH` is also an environment variable. Should we set this env variable here as well or property is sufficient?

Reply:
resolving this conversation